code/crates/engine/src/sync.rs

```diff
@@ -97,6 +97,9 @@ pub enum Msg<Ctx: Context> {
 
     /// We received an invalid [`CommitCertificate`] from a peer
     InvalidCommitCertificate(PeerId, CommitCertificate<Ctx>, CertificateError<Ctx>),
+
+    /// We received an invalid value from a peer
+    InvalidValue(PeerId, ProposedValue<Ctx>),
```

Comment:
Could we merge `InvalidCommitCertificate` and `InvalidValue`? Sync doesn't care why it is invalid. Only needs the `height` (not the full certificate or value) and `peer`.
Also have doubts about the `peer`, maybe sync should keep track of it. But we can discuss this later.